corehq/apps/geospatial/views.py

```diff
@@ -244,39 +284,62 @@ def get_paginated_cases_or_users_without_gps(request, domain):
     if case_or_user == 'user':
         data = _get_paginated_users_without_gps(domain, page, limit, query)
     else:
-        data = _get_paginated_cases_without_gps(domain, page, limit, query)
+        data = GetPaginatedCases(request, domain).get_paginated_cases_without_gps(domain, page, limit, query)
     return JsonResponse(data)
 
 
-def _get_paginated_cases_without_gps(domain, page, limit, query):
-    location_prop_name = get_geo_case_property(domain)
-    case_ids = (
-        CaseSearchES()
-        .domain(domain)
-        .is_closed(False)
-        .case_property_missing(location_prop_name)
-        .search_string_query(query, ['name'])
-        .sort('server_modified_on', desc=True)
-    ).get_ids()
-
-    paginator = Paginator(case_ids, limit)
-    case_ids_page = list(paginator.get_page(page))
-    cases = CommCareCase.objects.get_cases(case_ids_page, domain, ordered=True)
-    case_data = []
-    for case_obj in cases:
-        lat, lon = get_lat_lon_from_dict(case_obj.case_json, location_prop_name)
-        case_data.append(
-            {
-                'id': case_obj.case_id,
-                'name': case_obj.name,
-                'lat': lat,
-                'lon': lon,
-            }
+class GetPaginatedCases(CaseListMixin):
+    search_class = CaseSearchES
+
+    def __init__(self, request, domain, **kwargs):
+        # override super class corehq.apps.reports.generic.GenericReportView init method to
+        # avoid failures for missing expected properties for a report and keep only necessary properties
+        self.request = request
+        self.request_params = json_request(self.request.GET)
+        self.domain = domain
+
+    def _base_query(self):
+        # override CaseListMixin _base_query method to avoid pagination in ES and handle it later
+        return (
+            self.search_class()
+            .domain(self.domain)
         )
-    return {
-        'items': case_data,
-        'total': paginator.count,
-    }
+
+    def get_paginated_cases_without_gps(self, domain, page, limit, query):
+        show_cases_with_missing_gps_data = True
```

Comment:
Nit: Instead of requiring this flag, we can do the `show_all` check on the conditional where we're applying the filter:
```
cases_query = self._build_query()
location_prop_name = get_geo_case_property(domain)
if not GPSDataFilter(self.request, self.domain).show_all:
    cases_query = cases_query.case_property_missing(location_prop_name)
...
```

Reply:
Hey @zandre-eng 

yes, I did write that as well during my iterations.

The reason I went with what I did was to set up a default behavior that this page will show only missing gps data.
And then modify that based on anything else.
That is also true for the template since in options we are selecting the option in dropdown to show only default value, unless changed by the user.
So just context wise the view and the template start with the same setup.

Reply:
Thanks for the clarification. I'm happy to then keep this as is. :+1: